### PR TITLE
PN-413: Add information to the Reference and Matched columns to show institution/group that submitted the case

### PR DESCRIPTION
**Pull request by veronikaslc**
Done.

### CODE REVIEW

**Comment by sashaandjic** [comment]
In the attached screen shot I removed email address for John and Pat on purpose. 
Here are some suggestions that may improve the look and feel:

- change the format to display to:
  user name (email address)
  institution name

OR

- show email address when you hoover over envelope icon instead of displaying "Mark match contact to notify". The current tooltip is very confusing to the user anyway.

image

@marta wdyt?

File: matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/PatientInMatch.java [hunk 1]
```diff
@@ -132,11 +133,6 @@
      */
     Patient getPatient();
 
-    /**
-     * @return patient href.
-     */
-    String getHref();
```
Comment (allasm) [inline matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/PatientInMatch.java]: As discussed, for existing MME matches there may be data in the `href` column which is  not present anywhere else, so it is important not to lose it and display it as we used to display before this change

Comment (allasm) [reply matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/PatientInMatch.java]: we can either write a migrator or (less work) just use the data from the column if it is available

File: matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java [hunk 2]
```diff
@@ -259,20 +261,18 @@ public boolean isLocal()
...
-                // MME emails may be of the form "mailto:email1,email2", need ot parse that
-                if (this.href.startsWith("mailto:")) {
-                    String emailList = this.href.replace("mailto:", "");
+                // MME emails may be of the form "mailto:email1,email2", need to parse that
+                String href = this.contactInfo.getUrl();
+                if (href.startsWith("mailto:")) {
+                    String emailList = href.replace("mailto:", "");
                     emails.addAll(Arrays.asList(emailList.split(",|;")));
                 } else {
-                    emails.add(this.href);
```
Comment (allasm) [inline matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java]: this is the line which used that piece of data (...and which used to be transferred to front-end as one of the emails)

File: matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java [hunk 3]
```diff
@@ -258,25 +260,10 @@ public boolean isLocal()
...
-                    emails.addAll(Arrays.asList(emailList.split(",|;")));
-                } else {
-                    emails.add(this.href);
-                }
-            }
+        if (this.contactInfo != null) {
+            return this.contactInfo.getEmails();
         }
-        return emails;
+        return null;
```
Comment (allasm) [inline matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java]: there is a change in functionality, it used to return an empty collection when there were no emails, now it returns `null`. Null often requires an extra check in the calling code, any reason not to return an empty collection here as before?

File: matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java [hunk 4]
```diff
@@ -498,21 +493,42 @@ private Patient getLocalPatient()
...
         }
-        PatientData<ContactInfo> data = this.patient.getData("contact");
-        if (data != null && data.size() > 0) {
-            ContactInfo contact = data.get(0);
-            if (contact != null) {
-                this.href = contact.getUrl();
+        return null;
+    }
+
+    private ContactInfo rebuildContactInfo(JSONObject contact, String oldMMEContactData)
```
Comment (allasm) [inline matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java]: if this is called via line 193 where we may have a new version of a patient (via `this.patient = getLocalPatient();`) we need to get contact info from patient as well, as it is higher priority than stored data.

I.e. the code should be something like

```
if (patient != null) {
   this.populateContactInfo()
} else if (contact != null) {
...
}
```

**Pull request merged**